app/controllers/posts_controller.rb

```diff
@@ -18,7 +18,7 @@ def create
     @post = Post.new(post_params)
 
     if @post.save
-      redirect_to root_path
+      redirect_to root_path, notice: 'Post was successfully created'
     else
       render :new
```

Comment:
- [ ] Kindly would you please render a flash message when the Post is not saved. This will help the user to know the reason his post was not saved.
- [ ] Also make sure you initialize the `commentsCounter` and the `likesCounter` to 0 when the post is created. Hint :point_down: 

```rb  
 # Kindly this is just onty an example note that you can solve this the way you want.
  def create
    @post = Post.new(post_params)
    @post.commentsCounter = 0
    @post.likesCounter = 0

    if @post.save
      redirect_to root_path, notice: 'Post was successfully created'
    else
      render :new
    end
  end
```

- [ ] kindly solve n+1 query problem in your posts#index action and all other files to improve the performance and speed of your queries. Hint :point_down: 

```.rb
  def index
    @user = User.find(params['user_id'])
    @posts = @user.posts.includes(:comments)
  end
```

Reply:
Thanks for the hints, I didn't notice those issues at the start :laughing: